sea-orm-codegen/tests/compact/cake.rs

```diff
@@ -17,19 +17,12 @@ pub enum Relation {
     Fruit,
 }
 
-impl Related<super::fruit::Entity> for Entity {
-    fn to() -> RelationDef {
-        Relation::Fruit.def()
-    }
-}
-
-impl Related<super::filling::Entity> for Entity {
-    fn to() -> RelationDef {
-        super::cake_filling::Relation::Filling.def()
-    }
-    fn via() -> Option<RelationDef> {
-        Some(super::cake_filling::Relation::Cake.def().rev())
-    }
+#[derive(Copy, Clone, Debug, EnumIter, DeriveRelatedEntity)]
+pub enum RelatedEntity {
+    #[sea_orm(entity = "super::fruit::Entity", to = "Relation::Fruit.def()")]
+    Fruit,
+    #[sea_orm (entity = "super::filling::Entity", to = "super::cake_filling::Relation::Filling.def()", via = "Some(super::cake_filling::Relation::Cake.def().rev())")]
+    Filling
```

Comment:
I think we should keep generating the plain old `Related<T>`. With that in mind, if user pass in a `--seaography` flag to the `sea-orm-cli generate entity`, then, an additional `RelatedEntity` enum will be generated.

I imagine the derive attribute of each variant in the `RelatedEntity` could be... to take advantage of the existing `Related<T>` impl and avoid saying the same thing twice.

```rs
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelatedEntity)]
pub enum RelatedEntity {
    #[sea_orm (entity = "super::filling::Entity")]
    Filling
}
```